From: Simon McVittie <smcv@debian.org>
Date: Wed, 17 Jan 2018 15:19:12 +0000
Subject: tests/bootloader-entries-crosscheck: Use Python 3-friendly sorting

This is a little clearer than a strcmp()-style negative/zero/positive
return, and also works in Python 2.

Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: https://github.com/ostreedev/ostree/pull/1457
---
 tests/bootloader-entries-crosscheck.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/bootloader-entries-crosscheck.py b/tests/bootloader-entries-crosscheck.py
index 38e8e45..5faa548 100755
--- a/tests/bootloader-entries-crosscheck.py
+++ b/tests/bootloader-entries-crosscheck.py
@@ -38,8 +38,8 @@ def fatal(msg):
     sys.stderr.write('\n')
     sys.exit(1)
 
-def compare_entries_descending(a, b):
-    return int(b['version']) - int(a['version'])
+def entry_get_version(entry):
+    return int(entry['version'])
 
 def get_ostree_option(optionstring):
     for o in optionstring.split():
@@ -65,7 +65,7 @@ for fname in os.listdir(loaderpath):
             v = line[s+1:]
             entry[k] = v
         entries.append(entry)
-    entries.sort(compare_entries_descending)
+    entries.sort(key=entry_get_version, reverse=True)
 
 # Parse SYSLINUX config
 with open(syslinuxpath) as f:
